ergo-wallet/src/main/scala/org/ergoplatform/wallet/boxes/DefaultBoxSelector.scala

```diff
@@ -40,10 +40,30 @@ object DefaultBoxSelector extends BoxSelector {
       res += unspentBox
     }
 
-    def balanceMet = currentBalance >= targetBalance
+    /**
+      * Helper functions which checks whether enough ERGs collected
+      */
+    def balanceMet: Boolean = {
+      val diff = currentBalance - targetBalance
+
+      // We estimate how many ERG needed for assets in change boxes
+      val assetsDiff = currentAssets.size - targetAssets.size
```

Comment:
This estimation is not accurate in general. It works for the given test case, but this slightly changed one is still failing:

* inputBoxes:
box 1 with amount 1001000000 nanoerg and token A of value 2
box 2 with amount 1000000 nanoerg
* amoundToSpend: 1001000000 (exactly matching box 1)
* targetAssets: token A of value 1